src/twisted/python/_setup.py

```diff
@@ -96,6 +96,7 @@
         'pyasn1',
         'cryptography >= 0.9.1',
         'appdirs >= 1.4.0',
+        'pynacl >= 1.0.1',
```

Comment:
libsodium is definitely still the "less acceptable to upper management" crypto library, but its presence here is because OpenSSH already extensively uses crypto primitives from djb and OpenSSL doesn't (yet) support them. If you want to support these things in the near term you'll need to pull in pynacl or do it in pure Python (don't do that).